src/library/scala/compat/ClassValueCompat.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Scala (https://www.scala-lang.org)
+ *
+ * Copyright EPFL and Lightbend, Inc.
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.compat
+
+import scala.util.Try
+
+private[scala] abstract class ClassValueCompat[T] {
+
+  private def checkClassValueAvailability(): Boolean = Try {
+    Class.forName("java.lang.ClassValue", false, getClass.getClassLoader)
+    true
+  }.getOrElse(false)
+
+  private val instance = if (checkClassValueAvailability()) new JavaClassValue() else new FallbackClassValue()
```

Comment:
is JVM, and Android lazy enough not to error reading `java.lang.ClassValue` in the parents of `JavaClassValue` before executing `checkClassValueAvailability`?

Reply:
Yeah, fairly sure the class isn't loaded until it's first seen.